addons/mrp/models/mrp_production.py

```diff
@@ -1205,6 +1217,37 @@ def action_confirm(self):
         self.filtered(lambda mo: mo.state == 'draft').state = 'confirmed'
         return True
 
+    def _link_workorders_and_moves(self):
+        self.ensure_one()
+        if not self.workorder_ids:
+            return
+        workorder_per_operation = {workorder.operation_id: workorder for workorder in self.workorder_ids}
+        self.allow_workorder_dependencies = self.bom_id.allow_operation_dependencies
+        if self.allow_workorder_dependencies:
+            for workorder in self.workorder_ids:
+                workorder.blocked_by_workorder_ids = [Command.link(workorder_per_operation[operation_id].id) for operation_id in workorder.operation_id.blocked_by_operation_ids]
+        else:
+            previous_workorder = self.env['mrp.production']
+            for workorder in self.workorder_ids:
+                if previous_workorder and previous_workorder.operation_id.bom_id == workorder.operation_id.bom_id:
+                    workorder.blocked_by_workorder_ids = [Command.link(previous_workorder.id)]
+                previous_workorder = workorder
+        boms = set(workorder.operation_id.bom_id for workorder in self.workorder_ids)
+        last_workorder_per_bom = {bom: self.workorder_ids.filtered(lambda wo: wo.operation_id.bom_id == bom and not wo.needed_by_workorder_ids)[-1] for bom in boms}
+        for move in (self.move_raw_ids | self.move_finished_ids):
+            if move.operation_id:
+                move.write({
+                    'workorder_id': workorder_per_operation[move.operation_id].id
+                })
+            else:
+                bom = move.bom_line_id.bom_id if move.bom_line_id else self.bom_id
+                if bom not in boms:
+                    bom = self.bom_id
```

Comment:
```suggestion
                bom = move.bom_line_id.bom_id if move.bom_line_id and move.bom_line_id.bom_id in  
boms else self.bom_id
```
I'm not sure if you need () around `move.bom_line_id and move.bom_line_id.bom_id in  
boms `

I'm also wondering if you can have a different `move.bom_line_id.bom_id` != `self.bom_id` here except when move.bom_line_id == False